### PR TITLE
New graph time api

### PR DESCRIPTION
**Pull request by cquiroz**
we are getting the integration time and the graph in two separate calls. We may want to merge them into one instead

This PR adds an endpoint that takes signal to noise requirements, calculates the integration time and with that value produces the graph in one call.

This will reduce the amount of redraws on the client

### CODE REVIEW

File: modules/model/src/main/scala/lucuma/itc/SpectroscopyTimeAndGraphResult.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+case class SpectroscopyTimeAndGraphResult(
+  serverVersion:       String,
+  dataVersion:         String,
+  exposureTime:        TimeSpan,
+  exposures:           PosInt,
+  ccds:                NonEmptyList[ItcCcd],
+  charts:              NonEmptyList[ItcChart],
+  peakSNRatio:         SignalToNoise,
+  atWavelengthSNRatio: Option[SignalToNoise]
+) derives Encoder.AsObject
```
Comment (swalker2m) [inline modules/model/src/main/scala/lucuma/itc/SpectroscopyTimeAndGraphResult.scala:37]: Could/should this share anything with the normal spectroscopy result?

**Review by swalker2m — APPROVED** [review]
Looks great!

File: modules/client/shared/src/main/scala/lucuma/itc/client/ItcClient.scala [hunk 2]
```diff
@@ -55,11 +60,16 @@ object ItcClient {
...
+      specCache         <- ItcCache.simple[F, SpectroscopyIntegrationTimeInput, IntegrationTimeResult]
+      imgCache          <- ItcCache.simple[F, ImagingIntegrationTimeInput, IntegrationTimeResult]
+      graphCache        <-
         ItcCache.simple[F, OptimizedSpectroscopyGraphInput, OptimizedSpectroscopyGraphResult]
-      http       <- Http4sHttpClient.of[F, Unit](uri)(Async[F], Http4sHttpBackend(client), Logger[F])
+      timeAndGraphCache <-
+        ItcCache.simple[F,
+                        SpectroscopyIntegrationTimeAndGraphInput,
+                        SpectroscopyIntegrationTimeAndGraphResult
+        ]
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply modules/client/shared/src/main/scala/lucuma/itc/client/ItcClient.scala:71]: Probably, We are not using this cache on the client so I didn't worry about it

File: modules/client/shared/src/main/scala/lucuma/itc/client/SpectroscopyTimeAndGraphInput.scala [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  wavelength:         Wavelength,
+  signalToNoise:      SignalToNoise,
+  signalToNoiseAt:    Option[Wavelength],
+  sourceProfile:      SourceProfile,
+  band:               Band,
+  radialVelocity:     RadialVelocity,
+  constraints:        ConstraintSet,
+  mode:               InstrumentMode,
+  significantFigures: Option[SignificantFiguresInput]
+)
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply modules/client/shared/src/main/scala/lucuma/itc/client/SpectroscopyTimeAndGraphInput.scala:36]: hmm I see, we may need in the future to add parameters to override exp time but I guess that would fall in the same logic.
I'll update the schema as you suggest

File: modules/model/src/main/scala/lucuma/itc/SpectroscopyTimeAndGraphResult.scala (hunk 1 above)
Comment (cquiroz) [reply modules/model/src/main/scala/lucuma/itc/SpectroscopyTimeAndGraphResult.scala:37]: Right, we could extend one from the other and maybe encode it "correctly"

File: modules/client/shared/src/main/scala/lucuma/itc/client/ItcClient.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (swalker2m) [reply modules/client/shared/src/main/scala/lucuma/itc/client/ItcClient.scala:71]: Oh, it doesn't use the `itc-client`?  The cache is just a `Map` (not even an LRU) so it'll grow and grow until the process is restarted.

[the remark this answers is not in the document]
Comment (swalker2m) [reply modules/client/shared/src/main/scala/lucuma/itc/client/ItcClient.scala:71]: OTOH, the ODB won't be using this call so maybe it doesn't matter.

File: modules/model/src/main/scala/lucuma/itc/SpectroscopyTimeAndGraphResult.scala (hunk 1 above)
Comment (swalker2m) [reply modules/model/src/main/scala/lucuma/itc/SpectroscopyTimeAndGraphResult.scala:37]: Or just have the graph result include the spectroscopy result like the input.  Maybe it doesn't make sense.

**Comment by cquiroz** [comment]
It seems nested params are kind of involved. I'll implement the suggestion in another PR

**Pull request merged**